tensorflow/python/ops/rnn.py

```diff
@@ -611,22 +622,7 @@ def _dynamic_rnn_loop(cell,
   inputs_got_shape = tuple(input_.get_shape().with_rank_at_least(3)
                            for input_ in flat_input)
 
-  const_time_steps, const_batch_size = inputs_got_shape[0].as_list()[:2]
-
-  for shape in inputs_got_shape:
-    if not shape[2:].is_fully_defined():
-      raise ValueError(
-          "Input size (depth of inputs) must be accessible via shape inference,"
-          " but saw value None.")
-    got_time_steps = shape[0].value
-    got_batch_size = shape[1].value
-    if const_time_steps != got_time_steps:
```

Comment:
removing these things is weird; we're just checking consistency across multiple input tensors.  why do you need to remove these lines?  perhaps these ifs should only happen if const_time_steps and got_time_steps are both not None?